internal/xds/rbac/rbac_engine.go

```diff
@@ -238,3 +274,38 @@ type rpcData struct {
 	// handshake.
 	certs []*x509.Certificate
 }
+
+func (e *engine) doAuditLogging(rpcData *rpcData, rule string, authorized bool) {
+	// In the RBAC world, we need to have a SPIFFE ID as the principal for this
+	// to be meaningful
+	principal := ""
+	if len(rpcData.certs) != 0 && len(rpcData.certs[0].URIs) != 0 && rpcData.certs[0].URIs[0].Scheme == "spiffe" {
```

Comment:
Just realized in Go you need to use the [TLSInfo](https://github.com/grpc/grpc-go/blob/5dcfb37c0b43586965ed1ffb86bc63d706bc2c4e/credentials/tls.go#L35) to get Spiffe ID instead of trying to extract from the cert here. Please check the `AuthType()` of `peer.Peer.AuthInfo` and only populate Spiffe ID as the principal if the type is `tls`.

Reply:
`SPIFFEID` should have been validated by https://github.com/grpc/grpc-go/blob/8eba9c2de14ab211f21633104165fb082d272bfe/internal/credentials/spiffe.go#L45, you may remove the scheme check here. Feel free to resolve this comment afterwards.